src/frontend/mame/luaengine.cpp

```diff
@@ -1786,21 +1820,55 @@ void lua_engine::initialize()
 	image_type.set("is_reset_on_load", sol::property(&device_image_interface::is_reset_on_load));
 	image_type.set("must_be_loaded", sol::property(&device_image_interface::must_be_loaded));
 
+/*  device_slot_interface library
+ *
+ * manager:machine().slots[slot_name]
+ *
+ * slot.fixed - whether this slot is fixed, and hence not selectable by the user
+ * slot.has_selectable_options - does this slot have any selectable options at all?
+ * slot.default_option - returns the default option if one exists
+ * slot.options[] - get options table (k=name, v=device_slot_interface::slot_option)
+ */
+
+	auto slot_type = sol().registry().new_usertype<device_slot_interface>("new", sol::no_constructor);
+	slot_type["fixed"] = sol::property(&device_slot_interface::fixed);
+	slot_type["has_selectable_options"] = sol::property(&device_slot_interface::has_selectable_options);
+	slot_type.set("default_option", &device_slot_interface::default_option);
+	slot_type.set("options", sol::property([this](device_slot_interface &slot) {
+			sol::table option_table = sol().create_table();
+			for (auto &x : slot.option_list())
+				option_table[x.first] = &*x.second;
+			return option_table;
+		}));
+	sol().registry().set("slot", slot_type);
+
+
+/*  device_slot_interface::slot_option library
+ *
+ * manager:machine().slots[slot_name].options[option_name]
+ *
+ * slot_option:selectable()
+ */
+
+	auto dislot_option_type = sol().registry().new_usertype<device_slot_interface::slot_option>("new", sol::no_constructor);
+	dislot_option_type.set("selectable", &device_slot_interface::slot_option::selectable);
+	sol().registry().set("dislot_option", dislot_option_type);
```

Comment:
This is also using old syntax.

If you’re exposing this thing, why not also expose the shortname and description of the slot option’s device type, so the script can generate a list like `-listslots`?  You can expose the default BIOS and clock frequency (remember this can contain magic numbers for ratio of slot clock), too, for informational purposes.